File: app/src/main/java/com/pampang/nav/remote/RetrofitClient.java
package com.pampang.nav.remote;

import android.content.Context;

import com.pampang.nav.fcm.FCMAuthInterceptor;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {
    private static Retrofit fcmRetrofit = null;
    private static final String FCM_BASE_URL = "https://fcm.googleapis.com/";

    public static Retrofit getFCMClient(Context context) {
        if (fcmRetrofit == null) {
            OkHttpClient client = new OkHttpClient.Builder()
                    .addInterceptor(new FCMAuthInterceptor(context))
                    .build();

            fcmRetrofit = new Retrofit.Builder()
                    .baseUrl(FCM_BASE_URL)
                    .client(client)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return fcmRetrofit;
    }
}

File: app/src/main/java/com/pampang/nav/network/RetrofitClient.kt
package com.pampang.nav.network

import android.content.Context
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {

    private const val FCM_BASE_URL = "https://fcm.googleapis.com/"

    private var fcmRetrofit: Retrofit? = null

    fun getFCMClient(context: Context): Retrofit {
        if (fcmRetrofit == null) {
            val client = OkHttpClient.Builder()
                    .addInterceptor(FCMAuthInterceptor(context))
                    .build()

            fcmRetrofit = Retrofit.Builder()
                    .baseUrl(FCM_BASE_URL)
                    .client(client)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
        }
        return fcmRetrofit!!
    }
}
